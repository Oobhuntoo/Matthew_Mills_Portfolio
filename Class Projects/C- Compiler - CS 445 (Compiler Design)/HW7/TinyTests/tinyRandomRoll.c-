main()
{
    int i, sum;

    i=sum=0;
    while i<1000 do {
        sum += ?6;
        i++;
    }
    output(sum);  // about 2500
    outnl();

    outputb(?3 < 3);               // T
    outnl();
}
    
