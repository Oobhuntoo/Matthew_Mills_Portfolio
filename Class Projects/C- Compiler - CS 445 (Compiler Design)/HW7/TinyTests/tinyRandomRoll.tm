* C- compiler version C-S20
* Built: Apr 19, 2020
* Author: Robert B. Heckendorn
* File compiled:  tinyRandomRoll.c-
* 
* ** ** ** ** ** ** ** ** ** ** ** **
* FUNCTION input
  1:     ST  3,-1(1)	Store return address 
  2:     IN  2,2,2	Grab int input 
  3:     LD  3,-1(1)	Load return address 
  4:     LD  1,0(1)	Adjust fp 
  5:    LDA  7,0(3)	Return 
* END FUNCTION input
* 
* ** ** ** ** ** ** ** ** ** ** ** **
* FUNCTION output
  6:     ST  3,-1(1)	Store return address 
  7:     LD  3,-2(1)	Load parameter 
  8:    OUT  3,3,3	Output integer 
  9:    LDC  2,0(6)	Set return to 0 
 10:     LD  3,-1(1)	Load return address 
 11:     LD  1,0(1)	Adjust fp 
 12:    LDA  7,0(3)	Return 
* END FUNCTION output
* 
* ** ** ** ** ** ** ** ** ** ** ** **
* FUNCTION inputb
 13:     ST  3,-1(1)	Store return address 
 14:    INB  2,2,2	Grab bool input 
 15:     LD  3,-1(1)	Load return address 
 16:     LD  1,0(1)	Adjust fp 
 17:    LDA  7,0(3)	Return 
* END FUNCTION inputb
* 
* ** ** ** ** ** ** ** ** ** ** ** **
* FUNCTION outputb
 18:     ST  3,-1(1)	Store return address 
 19:     LD  3,-2(1)	Load parameter 
 20:   OUTB  3,3,3	Output bool 
 21:    LDC  2,0(6)	Set return to 0 
 22:     LD  3,-1(1)	Load return address 
 23:     LD  1,0(1)	Adjust fp 
 24:    LDA  7,0(3)	Return 
* END FUNCTION outputb
* 
* ** ** ** ** ** ** ** ** ** ** ** **
* FUNCTION inputc
 25:     ST  3,-1(1)	Store return address 
 26:    INC  2,2,2	Grab char input 
 27:     LD  3,-1(1)	Load return address 
 28:     LD  1,0(1)	Adjust fp 
 29:    LDA  7,0(3)	Return 
* END FUNCTION inputc
* 
* ** ** ** ** ** ** ** ** ** ** ** **
* FUNCTION outputc
 30:     ST  3,-1(1)	Store return address 
 31:     LD  3,-2(1)	Load parameter 
 32:   OUTC  3,3,3	Output char 
 33:    LDC  2,0(6)	Set return to 0 
 34:     LD  3,-1(1)	Load return address 
 35:     LD  1,0(1)	Adjust fp 
 36:    LDA  7,0(3)	Return 
* END FUNCTION outputc
* 
* ** ** ** ** ** ** ** ** ** ** ** **
* FUNCTION outnl
 37:     ST  3,-1(1)	Store return address 
 38:  OUTNL  3,3,3	Output a newline 
 39:     LD  3,-1(1)	Load return address 
 40:     LD  1,0(1)	Adjust fp 
 41:    LDA  7,0(3)	Return 
* END FUNCTION outnl
* 
* ** ** ** ** ** ** ** ** ** ** ** **
* FUNCTION main
 42:     ST  3,-1(1)	Store return address 
* COMPOUND
* Compound Body
* EXPRESSION
 43:    LDC  3,0(6)	Load integer constant 
 44:     ST  3,-3(1)	Store variable sum
 45:     ST  3,-2(1)	Store variable i
* WHILE
 46:     LD  3,-2(1)	Load variable i
 47:     ST  3,-4(1)	Save left side 
 48:    LDC  3,1000(6)	Load integer constant 
 49:     LD  4,-4(1)	Load left into ac1 
 50:    TLT  3,4,3	Op < 
 51:    JNZ  3,1(7)	Jump to while part 
* DO
* COMPOUND
* Compound Body
* EXPRESSION
 53:    LDC  3,6(6)	Load integer constant 
 54:    RND  3,3,6	Op ? 
 55:     LD  4,-3(1)	load lhs variable sum
 56:    ADD  3,4,3	op += 
 57:     ST  3,-3(1)	Store variable sum
* EXPRESSION
 58:     LD  3,-2(1)	load lhs variable i
 59:    LDA  3,1(3)	increment value of i
 60:     ST  3,-2(1)	Store variable i
* END COMPOUND
 61:    LDA  7,-16(7)	go to beginning of loop 
 52:    LDA  7,9(7)	Jump past loop [backpatch] 
