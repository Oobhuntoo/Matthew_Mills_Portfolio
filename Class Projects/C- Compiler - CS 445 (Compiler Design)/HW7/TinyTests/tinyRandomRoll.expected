Loading file: tinyRandomRoll.tm

Status: Halted
Bye.
